test/integration-new/regression-tests/cni_addon_test.go

```diff
@@ -0,0 +1,153 @@
+package regression_tests
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/aws/services"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	k8sUtils "github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/integration-new/common"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/apps/v1"
+)
+
+var (
+	err         error
+	podLabelKey = "app"
+	podLabelVal = "host-networking-test"
+	deployment  *v1.Deployment
+)
+
+const (
+	DEFAULT_VERSION = "v1.7.5-eksbuild.2"
+	LATEST_VERSION  = "v1.10.1-eksbuild.1"
```

Comment:
LATEST_VERSION can be set to default latest release which will be useful for daily or weekly runs. But when user runs this test, we need to provide an option for them to pass the built image (with their fix) or even better if we can build the image and pass it to this test if possible.

Reply:
Yes, there is an option to specify initial and target addons. If those are not provided then we fallback to these values.
We do want to be able to provide a custom built image url so I am working on a bash script which will invoke the ginkgo tests. Similar to 'run-canary-test.sh' that we have right now. 
Applying manifest files via ginkgo requires you to create every single resource in manifest, don't think there is readily available equivalent for 'kubectl apply'.So am working on bash script, while this is for published addons and can be treated like a regression test.

Reply:
Sounds good. Thanks.